frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/quickfix/CreateBranchQuickFix.java

```diff
@@ -0,0 +1,72 @@
+package com.virtuslab.gitmachete.frontend.file.quickfix;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+
+import java.util.Collections;
+
+import com.intellij.codeInsight.intention.IntentionAction;
+import com.intellij.codeInspection.util.IntentionFamilyName;
+import com.intellij.codeInspection.util.IntentionName;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.IncorrectOperationException;
+import git4idea.branch.GitBrancher;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import lombok.experimental.ExtensionMethod;
+
+import com.virtuslab.gitmachete.frontend.file.MacheteFileUtils;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitMacheteBundle.class})
+public class CreateBranchQuickFix implements IntentionAction {
+
+  private final String branch;
+  private final String parentBranch;
+  private final PsiFile macheteFile;
+
+  public CreateBranchQuickFix(String processedBranchName, String parentBranchName, PsiFile processedFile) {
+    branch = processedBranchName;
+    parentBranch = parentBranchName;
+    macheteFile = processedFile;
+  }
+
+  @Override
+  public @IntentionName String getText() {
+    return getNonHtmlString("action.GitMachete.BaseSlideInBelowAction.dialog.create-new-branch.title")
+        .format(parentBranch) + ": " + branch;
+  }
+
+  @Override
+  public @IntentionFamilyName String getFamilyName() {
+    return "Git Machete";
+  }
+
+  @Override
+  public boolean isAvailable(Project project, Editor editor, PsiFile file) {
+    return true;
+  }
+
+  @Override
+  @UIThreadUnsafe
```

Comment:
This is invoked outside of UI thread right?

Reply:
Yes. From `ivoke` javadoc:
> Called when user invokes intention. This method is called inside command. **If startInWriteAction() returns true**, this method is also called inside write action

And in `CreateBranchQuickFix` code we have:
```
public boolean startInWriteAction() {
    return false;
  }
```

Reply:
So, it is **not** called inside a write action. And write actions are run on Swing's EDT if I get it rigth.

Reply:
> If startInWriteAction() returns true, this method is also called inside write action

The implication is only in one direction: `startInWriteAction()` => call inside write action => we're on EDT

But it doesn't mean that the reverse is true ;) `startInWriteAction()` might be false and yet the code might execute on EDT (even if not within a write action) 🤔

Reply:
Ok, you are right - I checked that and in fact `SwingUtilities.isEventDispatchThread()` returns `true` for this method. Let me remove this `@UIThreadUnsafe` annotation.

Reply:
I removed these annotations, which results in following error from our checks:
```
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'methods that are not annotated with @UIThreadUnsafe and are not annotated with @IgnoreUIThreadUnsafeCalls should never call any @UIThreadUnsafe methods' was violated (1 times):
a non-@UIThreadUnsafe method com.virtuslab.gitmachete.frontend.file.quickfix.CreateBranchQuickFix.createNewBranchFromParent(com.intellij.openapi.project.Project) calls a @UIThreadUnsafe method com.virtuslab.gitmachete.frontend.file.MacheteFileUtils.findGitRepositoryForPsiMacheteFile(com.intellij.psi.PsiFile)
```
@PawelLipski Can we do something about it? Maybe disable this _violation_ for this one file?

Reply:
As `createNewBranchFromParent` calls `@UIThreadUnsafe` `MacheteFileUtils.findGitRepositoryForPsiMacheteFile`

Reply:
Ok, I just realised that I can fix this without such dirty tricks. Set `GitRepository gitRepository` in MacheteAnnotator and use it as another field in the `CreateBranchQuickFix` class

Reply:
Okay :) in general there's `@IgnoreUIThreadUnsafeCalls` for allowing UI-thread-unsafe calls from UI thread... but it's to be used sparingly (we've got it in like 3 places only so far).

Reply:
Yeah, I used this one